### DISCUSSION TITLE
TUI TREE - How to make an async load of a starting node?

### DISCUSSION THREAD

**Discussion by Makavto** [taiga-family/taiga-ui#5499] [Q&A] [answered]
In docs you have an example of async tree. There is a `TUI_TREE_START` `InjectionToken` that has value. But how can I make an async load of this `TUI_TREE_START`? 
I've tried an async function in `useFactory` and it didn't work. 
I also tried to get value via `APP_INITIALIZER` provider but `TUI_TREE_START` provider did not wait for the end of `APP_INITIALIZER` `useFactory` function.

**Comment by waterplea** [comment, marked as answer]
`TUI_TREE_START` is only requested in `TuiTreeService`, you can hide whatever injects `TuiTreeService` behind `ngIf` until you receive all the data you need.
